scratchapi2/front.py

```diff
@@ -16,93 +16,115 @@ class FrontPage
-        result = requests.get("https://api.scratch.mit.edu/proxy/featured").json()["community_newest_projects"]
-        projects = []
+            yield Project(item["id"], getinfo=GETINFO)
+
+    def new_projects(self):
+        """Get new Projects."""
+        warn("The recent projects row was removed. \
+This API endpoint may disappear in the future.",
+             PendingDeprecationWarning)
+        result = self._request('proxy/featured')["community_newest_projects"]
         for item in result:
-            projects.append(Project(item["id"], getinfo=False))
-        return projects
-
-    @staticmethod
-    def most_remixed():
-        """ Get Most Remixed Projects """
-        result = requests.get("https://api.scratch.mit.edu/proxy/featured").json()["community_most_remixed_projects"]
-        projects = []
+            yield Project(item["id"], getinfo=GETINFO)
+
+    def most_remixed_projects(self):
+        """Get most remixed Projects."""
+        result = self._request('proxy/featured')["community_most_remixed_projects"]
         for item in result:
-            projects.append(Project(item["id"], getinfo=False))
-        return projects
-
-    @staticmethod
-    def most_loved():
-        """ Get Most Loved Projects """
-        result = requests.get("https://api.scratch.mit.edu/proxy/featured").json()["community_most_loved_projects"]
-        projects = []
+            yield Project(item["id"], getinfo=GETINFO)
+
+    def most_loved_projects(self):
+        """Get most loved Projects."""
+        result = self._request('proxy/featured')["community_most_loved_projects"]
         for item in result:
-            projects.append(Project(item["id"], getinfo=False))
-        return projects
-
-    @staticmethod
-    def curator():
-        """ Get Curator's Selected Projects """
-        result = requests.get("https://api.scratch.mit.edu/proxy/featured").json()["curator_top_projects"]
-        projects = []
+            yield Project(item["id"], getinfo=GETINFO)
+
+    def curated_projects(self):
+        """Get the currently curated Projects and the current curator."""
+        result = self._request('proxy/featured')["curator_top_projects"]
         for item in result:
-            projects.append(GenericData(project=Project(item["id"], getinfo=False), curator=User(item["curator_name"], getinfo=False)))
-        return projects
-
-    @staticmethod
-    def sds():
-        """ Get SDS Projects """
-        result = requests.get("https://api.scratch.mit.edu/proxy/featured").json()["scratch_design_studio"]
-        projects = []
+            yield GenericData(
+               project=Project(item["id"], getinfo=GETINFO),
+               curator=User(item["curator_name"], getinfo=GETINFO)
+            )
+
+    def sds_projects(self):
+        """Get SDS Projects."""
+        result = self._request('proxy/featured')["scratch_design_studio"]
+        Studio = type('Studio', (GenericData, object), {

```

Comment:
I wonder if `Studio` should be a full-blown class, but there doesn't seem to be any API to fetch info about a studio.

Reply:
I had added it as a special case of GenericData, but since thisandagain* told me that there's no api to get an username from an ID, I removed it.

Some returns studio information, like studio search or Project.studios()